src/renderer/react/components/nodes/list/NodeList.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { nodeShape, nodeTreeElement } from 'react/shapes/node';
+import NodeListElement from 'react/components/nodes/listElement/NodeListElement';
+
+import classNames from './NodeList.module.scss';
+
+/**
+ * Builds the rendering tree of nodes.
+ *
+ * @param {object[]} treeElements - An array of nodes.
+ * @param {object} nodesMap - Nodes hashmap.
+ * @param {object} props - Props to propagate.
+ * @returns {object} - JSX render.
+ */
+const renderChildren = (treeElements, nodesMap, props) => (
```

Comment:
Why not just write it like 

```js
const NodeList = ({
  nodesTree,
  nodesMap,
  ...props
}) => (
  <ul className={classNames.listWrapper}>
    {treeElements.map(node => (
      <li key={node.id}>

        {/* Display the node itself. */}
        <NodeListElement
          node={nodesMap[node.id]}
          {...props}
        />

        {/* Then call the function recursively to render this node's children. */}
        {(node.children || []).length > 0
        && renderChildren(
          node.children,
          nodesMap,
          props,
        )}

      </li>
    ))}
  </ul>
);
```

Reply:
You forgot to render the component itself right ?